### ISSUE TITLE
Failure parsing host.json during zip push deploy causes silent failure in UI

### ISSUE THREAD

**Issue by paulbatum** [projectkudu/kudu#3153]
Ran into this issue when deploying a function app using the ZipDeployUI. The package I was deploying had a bad payload. Specifically, the host.json was missing a quote:

```json
{
  "version": "2.0",
  "logging": {
    "console": {
      "isEnabled": false
    },
    "fileLoggingMode": "never",
    "azureMonitorEnabled": false,
    "logLevel": {
      "default": "Error",
      "Host.Results": Information"
    }
  }  
}
```

The UI would get stuck in this state:
![image](https://user-images.githubusercontent.com/10663/80741271-38db1780-8ace-11ea-981a-b17297608671.png)

Eventually I realized that my deployment had not completed successfully, packagename.txt was never updated. I went looking in the traces:

```xml
<step title="BackgroundTrace" date="2020-04-30T17:28:33.039" instance="625e50" url="/api/zipdeploy/" method="POST" >
    <step title="Cleanup Xml Logs" date="2020-04-30T17:28:33.039" />
    <!-- duration: 0ms -->
    <step title="Creating temporary deployment" date="2020-04-30T17:28:33.054" />
    <!-- duration: 31ms -->
    <step title="Performing fetch based deployment" date="2020-04-30T17:28:33.085" >
        <step title="DeploymentManager.Deploy(id:03ea868fc2354b1b815559303475b5f4)" date="2020-04-30T17:28:33.132" >
            <step title="Collecting changeset information" date="2020-04-30T17:28:33.148" />
            <!-- duration: 31ms -->
            <step title="Updating submodules" date="2020-04-30T17:28:33.210" />
            <!-- duration: 31ms -->
            <step title="Determining deployment builder" date="2020-04-30T17:28:33.320" >
                <step title="Builder is RunFromZipSiteBuilder" date="2020-04-30T17:28:33.320" />
                <!-- duration: 0ms -->
            </step>
            <!-- duration: 16ms -->
            <step title="Copying SSH keys" date="2020-04-30T17:28:33.351" />
            <!-- duration: 16ms -->
            <step title="Building" date="2020-04-30T17:28:33.367" >
                <step title="Triggering recycle (preview mode disabled)." date="2020-04-30T17:28:33.476" />
                <!-- duration: 16ms -->
                <step title="Ignoring FS watcher based recycle request as this is a classic Windows app." date="2020-04-30T17:28:33.492" />
                <!-- duration: 0ms -->
                <step title="Purge deployment items" date="2020-04-30T17:28:33.601" >
                    <step title="Remove c59571a3c, Failed, received at 4/30/2020 5:21:02 PM" date="2020-04-30T17:28:33.648" />
                    <!-- duration: 0ms -->
                </step>
                <!-- duration: 47ms -->
            </step>
            <!-- duration: 281ms -->
            <step title="Error occurred" date="2020-04-30T17:28:33.648" type="error" text="Error parsing Infinity value. Path &apos;logging.logLevel[&apos;Host.Results&apos;]&apos;, line 11, position 25." stackTrace="   at Newtonsoft.Json.JsonTextReader.ParseNumberPositiveInfinity(ReadType readType)
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonTextReader.Read()
   at Newtonsoft.Json.Linq.JContainer.ReadContentFrom(JsonReader r, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JContainer.ReadTokenFrom(JsonReader reader, JsonLoadSettings options)
   at Newtonsoft.Json.Linq.JObject.Load(JsonReader reader, JsonLoadSettings settings)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at Kudu.Core.Helpers.PostDeploymentHelper.ReadDurableConfig(String hostConfigPath, Dictionary`2&amp; config) in C:\Kudu Files\Private\src\master\Kudu.Core\Helpers\PostDeploymentHelper.cs:line 376
   at Kudu.Core.Helpers.PostDeploymentHelper.&lt;SyncFunctionsTriggers&gt;d__30.MoveNext() in C:\Kudu Files\Private\src\master\Kudu.Core\Helpers\PostDeploymentHelper.cs:line 155
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Kudu.Core.Deployment.DeploymentManager.&lt;Build&gt;d__37.MoveNext() in C:\Kudu Files\Private\src\master\Kudu.Core\Deployment\DeploymentManager.cs:line 699" />
            <!-- duration: 16ms -->
        </step>
        <!-- duration: 531ms -->
        <step title="Cleaning up temp files" date="2020-04-30T17:28:33.664" />
        <!-- duration: 0ms -->
        <step title="WebHooksManager.PublishEventAsync: PostDeployment" date="2020-04-30T17:28:33.695" />
        <!-- duration: 47ms -->
    </step>
    <!-- duration: 703ms -->
</step>
<!-- duration: 766ms -->
```

Note the stacktrace showing a failure in synctriggers, specifically `Kudu.Core.Helpers.PostDeploymentHelper.ReadDurableConfig`.

I think there are two separate issues here:
- the UX should have indicated the error
- an invalid functions payload should still deploy. Errors in synctriggers should not abort the deployment

**Comment by suwatch** [comment]
We fix this #2713 to reportoverall deployment failure if synctrigger failed.   Portal should be able to see this from deployment center (one of deployment status would report failing).

**Comment by jvano** [comment]
Hi

If the problem persists and is related to running it on Azure App Service, please open a support incident in Azure:
https://learn.microsoft.com/en-us/azure/azure-portal/supportability/how-to-create-azure-support-request

This way we can better track and assist you on this case

Thanks,

Joaquin Vano
Azure App Service
